### PR TITLE
28 backend for creating an authenticating user

### PR DESCRIPTION
**Pull request by fadeomar**
#### Issues this PR relates to
closes : #28
=======

_NOTE: REMEMBER NOT TO REVIEW UNLESS ALL ACCEPTANCE CRITERIA HAS BEEN TICKED OFF_

---

#### Any info the reviewer needs to know to test 
e.g. Add env vars, run npm install

---

#### Any info the team needs to consider before merged into development
e.g. Will a database need to be rebuilt (and is this ok if existing data), add env vars

---

#### Quick Pull Request Checklist

Please make sure you ensure the following has been done before marking your PR as awaiting-review. For full checklist and process review, please refer to Process Guide (#8)

- Branch is up to date with develop
- Project runs
- You can download and install locally, and run the project locally
- All existing and new features run without bugs
- Checked that features/design works across all agreed browsers + devices
- The new work has been checked with Axe accessibility checker and doesn't throw significant errors  
- Linter is used
- Code is consistent with agreed team style
- Naming of files is consistent with project 
- Tests all pass
- No console logs

### CODE REVIEW

File: client/src/helpers/unique-id.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+function uniqueID() {
```
Comment (RamyAlshurafa) [inline client/src/helpers/unique-id.js:1]: also rename the file, and its calls
```suggestion
function createUniqueID() {
```

File: client/src/App.js [hunk 2]
```diff
@@ -7,138 +7,144 @@ import * as Pages from './pages';
...
-            />
-
-            <Route
-              exact
-              path={navRoutes.ADMIN.PENDING_ARTICLE}
-              Component={Pages.ArticlePage}
-              layout="general"
-              maxWidth="1050px"
-            />
+        <AuthProvider>
```
Comment (RamyAlshurafa) [inline client/src/App.js:19]: don't wrap everything in authProvider, just put it around the protected routes

**Review by RamyAlshurafa — CHANGES_REQUESTED** [review]
- [ ] sign up / login pages shouldn't be accessible by logged in users
- [ ] in the issue it says "Whenever they go to protected pages, on first app load check if they have pin in cookie and **check against Baserow** for pin and approved"

File: client/src/helpers/unique-id.js (hunk 1 above)
Comment (fadeomar) [reply client/src/helpers/unique-id.js:1]: done

File: client/src/components/Route/index.js [hunk 3]
```diff
@@ -1,14 +1,19 @@
...
+  const { user } = useAuth();
+  if (isPrivet && !user?.id) {
+    return <Redirect to={GENERAL.LOGIN} />;
+  }
   return (
-    <RouterRoute path={path} props exact={exact}>
+    <RouterRoute path={path} {...props} exact={exact}>
       <Layout layout={layout} {...props}>
-        <Component {...props} />
+        <Component {...props} user={user} />
```
[the remark this answers is not in the document]
Comment (fadeomar) [reply client/src/components/Route/index.js:16]: done

File: client/src/App.js (hunk 2 above)
Comment (fadeomar) [reply client/src/App.js:19]: done

File: client/src/context/auth.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
+import { createContext, useState, useContext } from 'react';
+
+const initialUserState = {
+  id: null,
+  email: '',
+  full_name: '',
```
[the remark this answers is not in the document]
Comment (fadeomar) [reply client/src/context/auth.js:6]: done

**Review by RamyAlshurafa — APPROVED** [review]

**Pull request merged**